java/google/registry/model/billing/RegistrarCreditBalance.java

```diff
@@ -49,6 +49,7 @@
  * taking the balance object with the latest effective time that is before (before or at) T, and
  * breaking any ties by choosing the mostly recently written among those balances.
  */
+@ReportedOn
```

Comment:
Have you actually built this PR and run the tests?  There are many errors that prevent it from building properly.  For example, each usage of the ReportedOn annotation needs to have a corresponding import statement.